### DISCUSSION TITLE
Rose edit LFRic-Apps

### DISCUSSION THREAD

**Discussion by dennissergeev** [MetOffice/simulation-systems#223] [LFRic]
This is probably a very trivial question but how do I open an LFRic-Apps rose-stem tests in the rose GUI with the correct metadata?

I've tried typing this in the `rose-stem` directory:
```bash
rose edit -M ../applications/lfric_atm/rose-meta/lfric-lfric_atm/HEAD
```
but when I click on the relevant app, I get an error "Could not find metadata for..."

**Comment by james-bruten-mo** [comment]
Hi Dennis,

The issue is that it can't find some of the metadata that it wants to import. LFRic Atm uses metadata from elsewhere in apps and also from core.

There are two options, One is to set the ROSE_META_PATH environment variable before running the edit command - this needs to be a comma separated list of paths, one to a core working copy and one to apps. eg. I've just run it as `export ROSE_META_PATH=/tmp/persistent/coreHead:/tmp/persistent/appsHead`.
Or you can use the -M option and specify the two paths, eg. `rose edit -M /tmp/persistent/coreHead -M /tmp/persistent/appsHead`.

Hopefully that works.

**Reply by dennissergeev** [reply]
Thanks @james-bruten-mo but I forgot to mention I'm doing this on Monsoon, where I'm guessing these metadata paths are different. If they are available, could you please point me to them?

**Reply by t00sa** [reply]
Hi @dennissergeev, the paths in James's post are examples.  You'll need to check out your own copies of the LFRic core and apps repositories on Monsoon and reference them with either the `-M` option or the `$ROSE_META_PATH` variable.

**Reply by dennissergeev** [reply]
Ah of course! Thank you @t00sa!

**Comment by iboutle** [comment]
As an aside to this, for the UM, there is a default ROSE_META_PATH setup, such that if you just run "rose edit" it finds the correct metadata from some central location. Is it possible to do the same for lfric_apps please?

**Reply by jennyhickson** [reply]
Hi Ian. The difference here is that Apps needs to know where the metadata is for both LFRic Apps and LFRic Core which makes it a bit clunkier. Hopefully this will improve with time, but for now it requires this more manual step. Cheers
